packages/components/src/components/Dropzone/Dropzone.tsx

```diff
@@ -0,0 +1,253 @@
+) => {
+  let icon: IconProps["icon"] = "CloudArrowUpIcon";
+  let iconColor: IconProps["color"] = "colorIconInfo";
+  if (isLoading || isDragAccept) {
+    icon = "CloudArrowUpIcon";
+    iconColor = "colorIconInfo";
+  }
+  if (successState && !isDragActive) {
+    icon = "CheckCircleIcon";
+    iconColor = "colorIconSuccess";
+  }
+  if (isDragReject || (errorState && !isDragActive)) {
+    icon = "ExclamationTriangleIcon";
+    iconColor = "colorIconError";
+  }
+
+  return (
+    <Box.div paddingBottom="space20">
+      <Icon color={iconColor} decorative icon={icon} size={"sizeIcon40"} />
+    </Box.div>
+  );
+};
+
+const Dropzone = React.forwardRef<HTMLDivElement, DropzoneProps>(
+  (
+    {
+      error,
+      fileTypes = { "application/pdf": [".pdf"] },
+      maxFileSize = MAX_FILE_SIZE,
+      onCancel,
+      onDrop,
+      progress,
+    },
+    ref
+  ) => {
+    const [dropZoneErrors, setDropZoneErrors] = useState(error);
+    const defaultState = !dropZoneErrors && !progress;
+    const errorState = dropZoneErrors;
+    const isLoading = !!progress && progress > 0 && progress < 100;
+    const successState = progress === 100 && !dropZoneErrors;
+
+    const {
+      acceptedFiles,
+      getRootProps,
+      getInputProps,
+      isDragActive,
+      isDragAccept,
+      isDragReject,
+      isFocused,
+    } = useDropzone({
+      maxFiles: 1,
+      maxSize: maxFileSize,
+      multiple: false,
+      accept: {
+        ...fileTypes,
+      },
+      onDropAccepted: (acceptedFiles) => {
+        setDropZoneErrors(undefined);
+        onDrop(acceptedFiles);
+      },
+      onDropRejected: (fileRejections) => {
+        forEach(fileRejections, (file) => {
+          forEach(file.errors, (err) => {
+            if (err.code === "file-too-large") {
+              setDropZoneErrors("File must be less than 50mb.");
+            }
+            if (err.code === "file-invalid-type") {
+              setDropZoneErrors(FileTypeErrorMessage);
+            }
+          });
+        });
+      },
+    });
+
+    return (
+      <>
+        {/*  // Ignoring this ts-error because getRootProps is typed as
+        // DropzoneRootProps which has a conflicting color prop. DropzoneRootProps
+        // comes directly from React-Dropzone so we have no control over it.
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        @ts-ignore */}
+        <Box.div
+          {...getColors(
+            errorState,
+            isLoading,
+            successState,
+            isFocused,
+            isDragAccept,
+            isDragReject
+          )}
+          alignItems="center"
+          borderRadius="borderRadius30"
+          borderStyle="borderDashed"
+          borderWidth="borderWidth10"
+          display="flex"
+          flexDirection="column"
+          padding="space70"
+          {...getRootProps()}
+          ref={ref}
+        >
+          {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
```

Comment:
Nit: Would you mind adding a comment to this one so we know why we're disabling it.